### ISSUE TITLE
Gitlab AutoDevOps with Kind

### ISSUE THREAD

**Issue by bilal-jafri** [kubernetes-sigs/kind#1615]
I created Kind cluster with following config file and was connected to private hosted gitlab. 
```apiVersion: kind.x-k8s.io/v1alpha4
kind: Cluster
networking:
  apiServerAddress: "X.X.X.X"
# add to the apiServer certSANs the name of the docker (dind) service in order to be able to reach the cluster through it
kubeadmConfigPatchesJSON6902:
  - group: kubeadm.k8s.io
    version: v1beta2
    kind: ClusterConfiguration
    patch: |
      - op: add
        path: /apiServer/certSANs/-
        value: docker
nodes:
  - role: control-plane
    kubeadmConfigPatches:
    - |
      kind: InitConfiguration
      nodeRegistration:
        kubeletExtraArgs:
          node-labels: "ingress-ready=true"
    extraPortMappings:
    - containerPort: 80
      hostPort: 80
      protocol: TCP
      listenAddress: "X.X.X.X"
    - containerPort: 443
      hostPort: 443
      protocol: TCP
      listenAddress: "X.X.X.X"
  - role: worker

```
After that I successfully Integrated my cluster and installed applications including helm tiller, Ingress and gitlab runner. The first problem occurred was in the displaying of Ingress Endpoint IP which keeps on loading
![image](https://user-images.githubusercontent.com/50214316/82442449-f2ab1f80-9a9f-11ea-9d96-c3d7fe7a64d0.png)

Meanwhile I tried to debug it and on Inspection it was found gitlab-managed-app-ingress external Ip was remaining pending. I tried to install Ingress manually with node port and also updated the Auto-DevOps template on gitlab and forcefully changed the ingress endpoints with following line in gitlab-ci.yml.

```
variables:
  KUBE_INGRESS_BASE_DOMAIN: X.X.X.X
```

After that the pipeline went successful until the deployment on staging. It fails on staging with error

![image](https://user-images.githubusercontent.com/50214316/82442929-e1164780-9aa0-11ea-9e6e-2e794ed93a8e.png)

Can someone suggest me a little bit on this? Correct me If I have done something wrong and stupid. Thanks in advance

**Comment by BenTheElder** [comment]
Does your kubectl version match the cluster version?

**Comment by bilal-jafri** [comment]
> Does your kubectl version match the cluster version?

My kubectl client version is v1.18.0

**Comment by BenTheElder** [comment]
looked closer at your output, gitlab or your tools really need to move to Deployment in apps/v1, which has been available since kubernetes 1.9

https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/
